src/pubsub/ua_pubsub.h

```diff
@@ -86,6 +89,26 @@ typedef struct UA_DataSetWriterSample{
 } UA_DataSetWriterSample;
 #endif
 
+typedef enum {
+    UA_PUBSUB_DATASETMESSAGE_TIMESTAMP,
+    UA_PUBSUB_DATASETMESSAGE_WRITERID,
```

Comment:
The writer id will not change once the WriterGroup is "frozen".
Why would we ever have to change it?

Reply:
Yes, you are right. The code to generate the offset structures are currently not implemented and i tried to use well known header fields for example.